scripts/run_smoke_tests.py

```
@@ -1158,6 +1158,98 @@ def __exit__(self, exc_type: object, exc: object, traceback: object) -> None:
             proxy_source.unlink(missing_ok=True)
 
+    class FakeProxyForHttpsTunnel:
+        def __init__(self) -> None:
+            self.connect_request: str | None = None
+            self.server: http.server.ThreadingHTTPServer | None = None
+            self.thread: threading.Thread | None = None
+
+        @property
+        def url(self) -> str:
+            if self.server is None:
+                raise RuntimeError("Fake proxy server is not running")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)